### PR TITLE
Adds job cancel methods, route and template UI

### PR DESCRIPTION
**Pull request by btylerburton**
# Pull Request

Related to:
- https://github.com/GSA/data.gov/issues/4761
- https://github.com/GSA/data.gov/issues/4866

## PR TASKS

- [X] The actual code changes.
- [X] Tests written and passed.
- [ ] Any changes to docs?

### CODE REVIEW

File: app/templates/view_job_data.html [hunk 1]
```diff
@@ -27,7 +27,18 @@ <h2>Job Info</h2>
...
+        <ul class="usa-button-group">
+            <li class="usa-button-group__item">
+                <a href="{{ url_for('harvest.cancel_harvest_job', job_id=data.harvest_job_id)}}">
+                    <button class="usa-button usa-button--secondary">Cancel</button>
+                </a>
+            </li>
+        </ul>
+    </div>
+    {% endif %}
+    <div class="section my-3">
```
Comment (jbrown-xentity) [inline app/templates/view_job_data.html:41]: This got renamed from mb-3 to my-3, is that intentional?

**Comment by btylerburton** [comment]
Fail to deploy to staging expected as there is no app in staging 😄

**Review by jbrown-xentity — APPROVED** [review]
LGTM, one question!

File: app/templates/view_job_data.html (hunk 1 above)
Comment (btylerburton) [reply app/templates/view_job_data.html:41]: yes, it's shorthand for `margin-bottom-3` -> `margin-y-3` for both top and bottom

**Pull request merged**